test/ntp_client_test.rb

```diff
@@ -0,0 +1,487 @@
+require_relative "test_helper"
+
+Yast.import "NtpClient"
+Yast.import "NetworkInterfaces"
+Yast.import "PackageSystem"
+Yast.import "Service"
+
+describe Yast::NtpClient do
+
+  subject { Yast::NtpClient }
+
+  describe "#Read" do
+    before do
+      subject.config_has_been_read = false
+      allow(Yast::NetworkInterfaces).to receive(:Read)
+      allow(subject).to receive(:Abort).and_return(false)
+      allow(Yast::Mode).to receive(:normal).and_return(false)
+      allow(Yast::Progress).to receive(:set)
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_NTP_POLICY"))
+        .and_return(nil)
+      allow(Yast::PackageSystem).to receive(:CheckAndInstallPackagesInteractive)
```

Comment:
I would avoid mocking with `allows` those method calls that are not used (almost) always. I prefer using `context` for that.